### PR TITLE
Feature/himitsu client cleanup

### PR DESCRIPTION
**Pull request by zonotope**
This branch contains all the commits from the feature/client-himitsu-client2 branch at the time of opening along with some cleanup including
- deleting the obsoleted `client` module and renaming the new `client2` module to `client`
- renaming the `gossip4` package prefix to `gossip`
- referencing a similarly cleaned up version of tupelo

### CODE REVIEW

**Comment by tobowers** [comment]
I cleaned up for the linter and turned off linting on gossip3

**Comment by tobowers** [comment]
now I've updated the go version for the github actions since using `fmt.Errorf("%w")` directives

**Comment by tobowers** [comment]
ok cool - passing tests.

File: gossip/remote/pubsubsimulator.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+		sb.eventStream.Publish(&simulatorMessage{
+			topic: topic,
+			msg:   message,
+		})
+	}
+
+	return nil
+}
+
+// returns subscriber props that can be used to listent to broadcast events
```
Comment (cap10morgan) [inline gossip/remote/pubsubsimulator.go]: extra `t` on `listen`

File: gossip/remote/pubsubsimulator.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+}
+
+func (bs *simulatedSubscriber) Receive(actorContext actor.Context) {
+	switch actorContext.Message().(type) {
+	case *actor.Started:
+		bs.Log.Debugw("subscribed", "topic", bs.topic, "subscribers", bs.subscribers)
+		parent := actorContext.Parent()
+		sub := bs.pubsubSystem.eventStream.Subscribe(func(evt interface{}) {
+			// there is a short delay on adding the predicate, so this make sure
+			// nothijng slips through
```
Comment (cap10morgan) [inline gossip/remote/pubsubsimulator.go]: typos: `s/make/makes/` & extra `j` snuck into `nothing`

File: gossip/remote/remote_test.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+	})
+
+	t.Run("sending a traceable when tracing is on", func(t *testing.T) {
+		// We *should* have serializable and traceable messages, but we don't.
+		// will take some work in the messages library (probaby with gogo protobuf)
+		// in order to bring those back. Without those, it's really hard to test this functionality.
+		t.Skip("we no longer have any serializable, and traceable messages at the moment.")
+	})
+
+	t.Run("when the otherside is closed permanently", func(t *testing.T) {
```
Comment (cap10morgan) [inline gossip/remote/remote_test.go]: space between `other` and `side`?

File: gossip/remote/router.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+		plugin.Use(&middleware.LogPlugin{}),
+	)
+}
+
+func (r *router) Receive(context actor.Context) {
+	// defer func() {
+	// 	if re := recover(); re != nil {
+	// 		r.Log.Errorw("recover", "re", re)
+	// 	}
+	// }()
```
Comment (cap10morgan) [inline gossip/remote/router.go]: still needed? should we add a comment about it?

File: gossip/client/client.go [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+	"github.com/quorumcontrol/tupelo-go-sdk/gossip/hamtwrapper"
+	"github.com/quorumcontrol/tupelo-go-sdk/gossip/types"
+	"github.com/quorumcontrol/tupelo-go-sdk/p2p"
+)
+
+const transactionTopic = "g4-transactions"
+
+// How many times to attempt PlayTransactions before giving up.
+// 10 is the library's default, but this makes it explicit.
+// var MaxPlayTransactionsAttempts = uint(10)
```
[the remark this answers is not in the document]
Comment (zonotope) [reply gossip/client/client.go]: we don't use it anymore, so I'm getting rid of it all together.

**Pull request merged**